core/data/src/main/java/org/the_chance/honeymart/data/source/local/AuthDataStorePreferencesImp.kt

```diff
@@ -40,4 +42,28 @@ class AuthDataStorePreferencesImp @Inject constructor(context: Context) : AuthDa
             preferences.remove(KEY_TOKEN)
         }
     }
+
+    override suspend fun saveProfileName(name: String) {
+        prefDataStore.edit { preferences ->
+            preferences[PROFILE_NAME] = name
+        }
+    }
+
+    override fun getProfileName(): String? {
```

Comment:
If `getProfileName` function needs to work on another thread wouldn't it be better to sue it a suspend